ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -54,12 +56,14 @@ public async Task<IActionResult> OnGetAsync(int? pageIndex)
                 nonDraftsIQ = service.FilterById(nonDraftsIQ, SearchParameters.IdFilter);
             }
 
-            IQueryable<Notification> notificationsIQ = service.OrderQueryableByNotificationDate(draftsIQ).Union(service.OrderQueryableByNotificationDate(nonDraftsIQ));
+            IQueryable<Notification> notificationIdsIQ = service.OrderQueryableByNotificationDate(draftsIQ).Union(service.OrderQueryableByNotificationDate(nonDraftsIQ));
 
-            var notifications = await PaginatedList<Notification>.CreateAsync(
-                notificationsIQ.AsNoTracking(), pageIndex ?? 1, pageSize);
+            var notificationIds = await service.GetPaginatedIdsAsync(notificationIdsIQ.Select(n => n.NotificationId), PaginationParameters);
+            var count = await notificationIdsIQ.CountAsync();
+            IEnumerable<Notification> notifications = await service.GetNotificationsWithDetailsById(notificationIds);
+            var paginatedList = new PaginatedList<Notification>(notifications, count, PaginationParameters);
 
-            SearchResults = notifications.SelectItems(NotificationBannerModel.WithLink);
+            SearchResults = paginatedList.SelectItems(NotificationBannerModel.WithLink);

```

Comment:
i think the `SelectItmes` is actually an overkill, now that you're feeding the `items` directly to the `PaginatedList` - you can just run a `Select` one step earlier :)